backend/src/services/git.ts

```diff
@@ -75,9 +75,20 @@ const setupGitConfig = (username: string, email: string, repositoryName: string)
 const gitCheckout = async (git:SimpleGit, branchName: string) => {
   const sanitizedBranchName = sanitizeBranchName(branchName)
   await validateBranchName(sanitizedBranchName)
-  await git.checkout([sanitizedBranchName])
+  
+  if (await branchExists(git, sanitizedBranchName)) {
+    await git.checkout([sanitizedBranchName])
+    return
+  }
+  
+  await git.checkout(['-b', sanitizedBranchName])
 }
 
+const branchExists = async (git:SimpleGit, branchName: string):Promise<boolean> => {
+  const branchSummaryResult = await git.branch(['--list', branchName])
```

Comment:
isn't the branchName parameter here used as pattern to check, if git branch --list includes branchName? If so, I think it makes iterating over Object.keys manually redundant